pkgs/applications/audio/jack-autoconnect/default.nix

```diff
@@ -0,0 +1,38 @@
+# Call with libsForQt5.callPackage
+
+{ lib, mkDerivation, fetchFromGitHub, pkg-config, qmake, qtbase, libjack2 }:
+
+let
+  # 1 February 2021
+  gitRevision = "fe0c8f69149e30979e067646f80b9d326341c02b";
+
+in mkDerivation rec {
+  pname = "jack_autoconnect";
+
+  # It does not have any versions (yet?)
+  version = "git-${builtins.substring 0 7 gitRevision}";
+
+  src = fetchFromGitHub {
+    owner = "kripton";
+    repo = "jack_autoconnect";
+    rev = gitRevision;
+    sha256 = "sha256-imvNc498Q2W9RKmiOoNepSoJzIv2tGvFG6hx+seiifw=";
+  };
+
+  buildInputs = [ qtbase libjack2 ];
+  nativeBuildInputs = [ pkg-config qmake ];
+
+  installPhase = ''
+    mkdir -p -- "$out/bin"
+    cp -- jack_autoconnect "$out/bin"
```

Comment:
```suggestion
    mkdir -p $out/bin
    cp jack_autoconnect $out/bin
```

Reply:
@winterqt but why?

Reply:
@unclechu They just seem extraneous to me, especially the `--`. If I may ask, is there a reason you added them, did the phase break unless they were there?

Reply:
Reason to have quotes almost always when writing shell scripts: Bash interpolates variables expanding values into multiple arguments if there are spaces. `$out` should be a single value. There is no reason to use it without quotes.

`--` is used because there are interpolated variables after it. There should be no options anymore but only positional arguments. Well, you kind of can give an argument that `$out` always starts with `/nix/store` and never with a dash. But even this I think is not totally guaranteed to hold true. I believe there are some options in Nix to have Nix Store placed somewhere else, maybe in a relative directory path? Could it be the case? This means technically directory name might start with a dash which will break the phase. I’m not sure, so I don’t see why should I intentionally change this line and take a risk. So if you keep `--` it will never go wrong like this. If you remove it you make an assumption what values will be, and that it’ll never change.

My mindset when writing Bash is to go safe by default, because Bash has bear traps everywhere. I’d rather give reasons why go unsafe than why go safe. Quotes and raw positional arguments separator is a default when interpolating variables. And there must be reasons to not use them.

Reply:
Fair enough, good points.